spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanDependencyResolverFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.spring.config.annotation;
+
+import org.axonframework.common.Priority;
+import org.axonframework.eventhandling.AnnotationEventHandlerAdapter;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.beans.factory.config.DependencyDescriptor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.MethodParameter;
+import org.springframework.lang.NonNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+
+/**
+ * ParameterResolverFactory implementation that resolves parameters using Spring dependency resolution.
+ * Mark a parameter as {@link org.springframework.beans.factory.annotation.Autowired} to resolve said parameter
+ * using Spring dependency resolution.
+ *
+ * @author Joel Feinstein
+ * @since 4.5
+ * @see Autowired
+ */
+@Priority(Priority.LOW)
```

Comment:
The priority of this guy should be higher than the soon-to-be-deprecated `SpringBeanParameterResolverFactory`. 
Otherwise, they'd be competing, whilst in essence, we want this to win and replace the old variant eventually.